packages/nextjs/test/config.test.ts

```diff
@@ -324,6 +327,45 @@ describe('webpack config', () => {
       );
     });
   });
+
+  describe('`distDir` value in default server-side `RewriteFrames` integration', () => {
+    describe('`DefinePlugin` existence and inclusion of `__rewriteFramesDistDir__` definition', () => {
+      it.each([
+        ['no custom `distDir`, pre-existing `DefinePlugin`', undefined, true],
+        ['no custom `distDir`, no pre-existing `DefinePlugin`', undefined, false],
+        ['custom `distDir`, pre-existing `DefinePlugin`', 'some/output/directory', true],
+        ['custom `distDir`, no pre-existing `DefinePlugin`', 'some/output/directory', false],
+      ])(
+        '%s',
+        async (_testTitle: string, customDistDir: string | undefined, existingDefinePlugin: boolean): Promise<void> => {
+          const mightHaveDistDir = {
+            ...userNextConfig,
+            ...(customDistDir && { distDir: customDistDir }),
+          };
+          const mightHaveDefinePlugin = {
+            ...serverWebpackConfig,
+            ...(existingDefinePlugin && { plugins: [new DefinePlugin({ __dogName__: 'Maisey' })] }),
+          };
+
+          const finalWebpackConfig = await materializeFinalWebpackConfig({
+            userNextConfig: mightHaveDistDir,
+            incomingWebpackConfig: mightHaveDefinePlugin,
+            incomingWebpackBuildContext: getBuildContext('server', mightHaveDistDir),
+          });
+
+          expect(finalWebpackConfig.plugins).toEqual(expect.arrayContaining([expect.any(DefinePlugin)]));
+
+          const definePluginInstance = findWebpackPlugin(finalWebpackConfig, 'DefinePlugin') as DefinePlugin;
+          expect(definePluginInstance.definitions.__rewriteFramesDistDir__).toEqual(customDistDir);
+        },
+      );
+    });
+    describe('`RewriteFrames` ends up with correct `distDir` value', () => {
+      // TODO: this, along with any number of other parts of the build process, should be tested with an integration
+      // test (and that integration test should test custom `distDir` values with and without a `.`, to make sure the
+      // regex escaping is working)
+    });
```

Comment:
Landing this PR has a risk of breaking users' builds. Why do think it's a `TODO`?

Reply:
Because it's an entire project akin to Kamil's integration tests. We need to figure out how to actually run the full build process inside of a test and see what comes out of it, and we also need to make sure we know exactly where to look in the output (given all of the code splitting, etc) that nextjs does as part of building). And while I don't disagree with you, that's true of every change we've made or will make to webpack config. 

All of that said, research I was doing just yesterday is giving me at least slight pause, specifically because of that code splitting. Let's look at this together IRL before we come back to this question.